### PR TITLE
[MDS-4737] major mines error final application save draft

### PR DESCRIPTION
**Pull request by areyeslo**
## Objective 

[MDS-4737](https://bcmines.atlassian.net/browse/MDS-4737)

- Ability to save a draft for a Major Mine Application
- Added Document Table: 
image

- Add validation in form incidents: 
image

### CODE REVIEW

File: services/minespace-web/src/routes/Routes.js [hunk 1]
```diff
@@ -33,6 +33,21 @@ const Routes = () => (
...
+    <Route
+      exact
+      path={routes.EDIT_MAJOR_MINE_APPLICATION.route}
+      component={AuthenticationGuard()(routes.EDIT_MAJOR_MINE_APPLICATION.component)}
+    />
+    <Route
+      exact
+      path={routes.ADD_MAJOR_MINE_APPLICATION.route}
+      component={AuthenticationGuard()(routes.ADD_MAJOR_MINE_APPLICATION.component)}
+    />
```
Comment (JustinMacaulay) [inline services/minespace-web/src/routes/Routes.js:50]: I know the routes were giving you trouble, did moving them up make a difference?

File: services/minespace-web/src/constants/routes.js [hunk 2]
```diff
@@ -102,6 +103,12 @@ export const MAJOR_MINE_APPLICATION_SUCCESS = {
   component: MajorMineApplicationSuccessPage,
 };
 
+export const ADD_MAJOR_MINE_APPLICATION = {
```
Comment (JustinMacaulay) [inline services/minespace-web/src/constants/routes.js:106]: Same here with the re-ordering.

File: services/minespace-web/src/components/pages/Project/MajorMineApplicationPage.js [hunk 3]
```diff
@@ -441,6 +468,7 @@ const mapDispatchToProps = (dispatch) =>
       reset,
       touch,
       change,
+      destroy,
```
Comment (JustinMacaulay) [inline services/minespace-web/src/components/pages/Project/MajorMineApplicationPage.js:471]: Do we use this anywhere else? I dont know that I have ever seen destroy.

**Review by JustinMacaulay — COMMENTED** [review]
Just a few comments

File: services/minespace-web/src/routes/Routes.js (hunk 1 above)
Comment (areyeslo) [reply services/minespace-web/src/routes/Routes.js:50]: Yes, the issue was related to the order of the routes. Sorting from more specific to more general plus adding `/edit` to Route for edit made the trick.

File: services/minespace-web/src/components/pages/Project/MajorMineApplicationPage.js (hunk 3 above)
Comment (areyeslo) [reply services/minespace-web/src/components/pages/Project/MajorMineApplicationPage.js:471]: Just in Incidents in minespace and core-web

File: services/minespace-web/src/routes/Routes.js (hunk 1 above)
Comment (areyeslo) [reply services/minespace-web/src/routes/Routes.js:50]: The routes for Major Mine Application are as follow:
/projects/:projectGuid/major-mine-application/:mmaGuid/success

/projects/:projectGuid/major-mine-application/:mmaGuid/review

/projects/:projectGuid/major-mine-application/:mmaGuid/edit

/projects/:projectGuid/major-mine-application/new

Before the edit Route was `/projects/:projectGuid/major-mine-application/:mmaGuid`. Somehow it was not found in the routes eventhough it was placed after review Route (keeping the most specific first, then more general). Then, the only way to make it work was to add `/edit` suffix to the route to make it similar to success and review.

Comment (JustinMacaulay) [reply services/minespace-web/src/routes/Routes.js:50]: Interesting but good to know.

**Review by BenHarrington — APPROVED** [review]
Thanks for digging in and getting those validation issues sorted out!

**Pull request merged**